pymc/backends/ndarray.py

```diff
@@ -115,6 +115,8 @@ def record(self, point, sampler_stats=None) -> None:
         if sampler_stats is not None:
             for data, vars in zip(self._stats, sampler_stats):
                 for key, val in vars.items():
+                    if isinstance(val, np.ndarray) and val.ndim > 0:
```

Comment:
Is the val always a single number?

Reply:
Right, currently `val` is converted to a scalar only when it is a single-element `ndarray`
I think the correct implementation would be 
```
if isinstance(val, np.ndarray):
    # Extract the first element using indexing (preserves order)
    val = val.flat[0]  
data[key][self.draw_idx] = val
```

Reply:
I'm not sure these records are supposed to always be single elements. If not, discarding the rest doesn't make sense